spec/unit/typesafe/config/tokenizer_spec.rb

```diff
@@ -0,0 +1,696 @@
+    let(:test_string) { "true foo" }
+    let(:expected_tokens) { [TestUtils.token_true,
+                             TestUtils.token_unquoted(" "),
+                             TestUtils.token_unquoted("foo")] }
+
+    include_examples "token_matching"
+  end
+
+  ####################
+  # Slashes
+  ####################
+  context "tokenize unquoted text containing slash" do
+    let(:test_string) { "a/b/c/" }
+    let(:expected_tokens) { [TestUtils.token_unquoted("a/b/c/")] }
+
+    include_examples "token_matching"
+  end
+
+  context "tokenize slash" do
+    let(:test_string) { "/" }
+    let(:expected_tokens) { [TestUtils.token_unquoted("/")] }
+
+    include_examples "token_matching"
+  end
+
+  context "tokenize slash space slash" do
+    let(:test_string) { "/ /" }
+    let(:expected_tokens) { [TestUtils.token_unquoted("/"),
+                             TestUtils.token_unquoted(" "),
+                             TestUtils.token_unquoted("/")] }
+
+    include_examples "token_matching"
+  end
+
+  ####################
+  # Quotes
+  ####################
+  context "tokenize mixed unquoted and quoted" do
+    let(:test_string) { "    foo\"bar\"baz   \n" }
+    let(:expected_tokens) { [TestUtils.token_unquoted("foo"),
+                             TestUtils.token_string("bar"),
+                             TestUtils.token_unquoted("baz"),
+                             TestUtils.token_line(1)] }
+
+    include_examples "token_matching"
+  end
+
+  context "tokenize empty triple quoted string" do
+    let(:test_string) { "\"\"\"\"\"\"" }
+    let(:expected_tokens) { [TestUtils.token_string("")] }
+
+    include_examples "token_matching"
+  end
+
+  context "tokenize trivial triple quoted string" do
+    let(:test_string) { "\"\"\"bar\"\"\"" }
+    let(:expected_tokens) { [TestUtils.token_string("bar")] }
+
+    include_examples "token_matching"
+  end
+
+  context "tokenize no escapes in triple quoted string" do
+    let(:test_string) { "\"\"\"\\n\"\"\"" }
+    let(:expected_tokens) { [TestUtils.token_string("\\n")] }
+
+    include_examples "token_matching"
+  end
+
+  context "tokenize trailing quotes in triple quoted string" do
+    let(:test_string) { "\"\"\"\"\"\"\"\"\"" }
+    let(:expected_tokens) { [TestUtils.token_string("\"\"\"")] }
+
+    include_examples "token_matching"
+  end
+
+  context "tokenize new line in triple quoted string" do
+    let(:test_string) { "\"\"\"foo\nbar\"\"\"" }
+    let(:expected_tokens) { [TestUtils.token_string("foo\nbar")] }
+
+    include_examples "token_matching"
+  end
+
+
+  ####################
+  # Find problems when tokenizing
+  ####################
+  context "problem nothing after backslash" do
+    let(:test_string) { " \"\\\" " }
+    include_examples "strings_with_problems"
+  end
+
+  context "problem there is no \q escape sequence" do
+    let(:test_string) { " \"\\q\" " }
+    include_examples "strings_with_problems"
+  end
+
+  context "problem too short" do
```

Comment:
It might be good to add a bit more to the description for each of these "problem too short" contexts to differentiate them from each other.